### PR TITLE
MAT-9555 add AdminShareLibrary export user list link

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-9555](https://jira.cms.gov/browse/MAT-9555)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

**Comment by sb-cecilialiu** [comment]
MAT-9555_share_exportLink

**Comment by sb-cecilialiu** [comment]
MAT-9555_unshare_exportLink

File: src/components/common/libraryShareDialog/LibraryShareDialog.tsx [hunk 1]
```diff
@@ -128,6 +130,8 @@ const LibraryShareDialog = ({
   const [confirmationDialogOpen, setConfirmationDialogOpen] = useState(false);
 
   const showShareDialog = option === "Share With" || option === "Unshare";
+  const isAdminShareLibraryEnabled =
+    useIsRoleOrFeatureEnabled("AdminShareLibrary");
```
Comment (RohitKandimalla) [inline src/components/common/libraryShareDialog/LibraryShareDialog.tsx:137]: What role are we checking here ? We are not passing in any roles here

File: src/components/common/libraryShareDialog/LibraryShareDialog.scss [hunk 2]
```diff
@@ -35,3 +35,37 @@
     margin-left: 20px;
   }
 }
+
+.export-link {
```
Comment (RohitKandimalla) [inline src/components/common/libraryShareDialog/LibraryShareDialog.scss]: Not sure why we want to use a button and style it like a link, while we can use Link directly

File: src/components/common/libraryShareDialog/LibraryShareDialog.tsx [hunk 3]
```diff
@@ -586,12 +608,16 @@ const LibraryShareDialog = ({
...
+            {option !== "Unshare" && getAdminUserExportLink()}
+          </div>
+          {option === "Unshare" && (
+            <div className="share-unshare-dialog-info-text">
               <div>
                 Deselect the users with whom you want to unshare the library(s).
               </div>
-            )}
-          </div>
+              {getAdminUserExportLink()}
```
Comment (RohitKandimalla) [inline src/components/common/libraryShareDialog/LibraryShareDialog.tsx]: 1. Why do we want this to be a function ?
2. We are calling the same function from 2 conditional blocks. Does it make sense to add it once at the end without any condition ? Since it should be displayed irrespective of the condition

File: src/components/common/libraryShareDialog/LibraryShareDialog.tsx [hunk 4]
```diff
@@ -520,6 +524,24 @@ const LibraryShareDialog = ({
     }
   };
 
+  const handleExportUserList = () => {
```
Comment (RohitKandimalla) [inline src/components/common/libraryShareDialog/LibraryShareDialog.tsx:530]: dead code. Probably we can remove it

File: src/components/common/libraryShareDialog/LibraryShareDialog.tsx (hunk 1 above)
Comment (sb-cecilialiu) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx:137]: @RohitKandimalla the role and the feature flag will be handled together within useIsRoleOrFeatureEnabled() in madie-util. As a matter of flag, my intention was to use it as the only interface when handling roles and/or feature flag, so we don't have to epxort/import feature flags every time a new one is added. There will be some refactor to achieve this, mainly in madie-measure. madie-cql-library has been done.

File: src/components/common/libraryShareDialog/LibraryShareDialog.scss (hunk 2 above)
Comment (sb-cecilialiu) [reply src/components/common/libraryShareDialog/LibraryShareDialog.scss]: @RohitKandimalla I guess Link is for navigation, and Button is more of action handling?

File: src/components/common/libraryShareDialog/LibraryShareDialog.tsx (hunk 4 above)
Comment (sb-cecilialiu) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx:530]: @RohitKandimalla Why is it dead? This one should trigger export user list file?

File: src/components/common/libraryShareDialog/LibraryShareDialog.tsx (hunk 3 above)
Comment (sb-cecilialiu) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx]: @RohitKandimalla 1. cleaner code. 2. Main reason is when adding it just once at the end, I couldn't display it right.

Comment (RohitKandimalla) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx]: Not really, we can achieve the same thing by not repeating the code instead use CSS to style them properly. I spoke with Prateek yesterday with the same w.r.t measure and advised him to make few updates. You can check my suggestion on his branch "MAT-9551-adminShareUnshareDialog"

File: src/components/common/libraryShareDialog/LibraryShareDialog.scss (hunk 2 above)
Comment (RohitKandimalla) [reply src/components/common/libraryShareDialog/LibraryShareDialog.scss]: Yes, for sure button suits this purpose. Then we can just say export-button instead of export-link then. Because after all it is not a link.

File: src/components/common/libraryShareDialog/LibraryShareDialog.tsx (hunk 4 above)
Comment (RohitKandimalla) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx:530]: It is not implemented yet. but I don't mind leaving it like that.

File: src/components/common/libraryShareDialog/LibraryShareDialog.tsx (hunk 1 above)
Comment (RohitKandimalla) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx:137]: Yes I understand the feature flag part of it. but the name of the hook says "IsRoleOrFeatureEnabled" What role is it talking about ? if the hook says "isFeatureFlagEnabled" and then you send a feature flag name, Then it definitely make sense.

Comment (sb-cecilialiu) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx:137]: @RohitKandimalla for a couple of featuer flags, the role is involved, e.g.: 

else if (feature === FeatureFlagsEnum.ADMIN_TRANSFER_LIBRARY) {
    return Boolean(featureFlags?.AdminTransferLibrary **&& userRoles?.isAdmin**);
  } else if (feature === FeatureFlagsEnum.ADMIN_SHARE_LIBRARY) {
    return Boolean(featureFlags?.AdminShareLibrary **&& userRoles?.isAdmin**);
  }

File: src/components/common/libraryShareDialog/LibraryShareDialog.scss (hunk 2 above)
Comment (sb-cecilialiu) [reply src/components/common/libraryShareDialog/LibraryShareDialog.scss]: @RohitKandimalla done.

File: src/components/common/libraryShareDialog/LibraryShareDialog.tsx (hunk 3 above)
Comment (sb-cecilialiu) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx]: @RohitKandimalla done

File: src/components/common/libraryShareDialog/LibraryShareDialog.tsx (hunk 1 above)
Comment (RohitKandimalla) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx:137]: So in ideal world it should be "useIsAdminRoleOrFeatureEnabled" but then the conditions in these specific roles is && and not OR

Comment (sb-cecilialiu) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx:137]: @RohitKandimalla For the admin share library functionality, it needs the feature flag AdminShareLibrary on AND the user role to be admin.

Comment (RohitKandimalla) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx:137]: Yup exactly and yet the function is called "useIsRoleOrFeatureEnabled"

Comment (RohitKandimalla) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx:137]: Should the function name be then "useIsAdminRoleAndFeatureEnabled" ?

Comment (sb-cecilialiu) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx:137]: @RohitKandimalla For this feature, it's an "AND", but for some other features, it's just the feature flag, that's why I used "OR". This interface was already done and in madie-utils.

Comment (RohitKandimalla) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx:137]: Yup unfortunately, I don't like the way it is implemented. But if it takes up too much time to make this better, then may be we can considering adding a tech debt to re-write the feature flag hook in madie-utils.

Comment (sb-cecilialiu) [reply src/components/common/libraryShareDialog/LibraryShareDialog.tsx:137]: @RohitKandimalla Can you explain why you don't like the way it is implemented? I thought it's better, because we don't have to export/import every time a new feature flag is added.

**Review by RohitKandimalla — APPROVED** [review]
